src/VisualStudio/Core/Def/Options/LocalUserRegistryOptionPersister.cs

```diff
@@ -29,12 +25,10 @@ private LocalUserRegistryOptionPersister(RegistryKey registryKey)
             _registryKey = registryKey;
         }
 
-        public static async Task<LocalUserRegistryOptionPersister> CreateAsync(IAsyncServiceProvider provider)
+        public static async Task<LocalUserRegistryOptionPersister> CreateAsync(IVsService<ILocalRegistry4> localRegistryService)
         {
             // SLocalRegistry service is free-threaded -- see https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1408594.
-            // Note: not using IAsyncServiceProvider.GetServiceAsync<TService, TInterface> since the extension method might switch to UI thread.
```

Comment:
To confirm the new version of this isn't reintroducing this bug?

Reply:
➡️ Correct. Our test implementation uses a Roslyn variation of `GetServiceAsync` which passes an explicit `JoinableTaskContext`.